src/main/java/com/osiris/autoplug/client/network/online/connections/SSHServerSetup.java

```diff
@@ -0,0 +1,134 @@
+/*
+ * Copyright (c) 2021-2023 Osiris-Team.
+ * All rights reserved.
+ *
+ * This software is copyrighted work, licensed under the terms
+ * of the MIT-License. Consult the "LICENSE" file for details.
+ */
+package com.osiris.autoplug.client.network.online.connections;
+
+import com.osiris.autoplug.client.configs.SSHConfig;
+import com.osiris.jlib.logger.AL;
+
+import org.apache.sshd.common.config.keys.AuthorizedKeyEntry;
+import org.apache.sshd.common.config.keys.KeyUtils;
+import org.apache.sshd.server.SshServer;
+import org.apache.sshd.server.auth.password.PasswordAuthenticator;
+import org.apache.sshd.server.auth.pubkey.PublickeyAuthenticator;
+import org.apache.sshd.server.command.Command;
+import org.apache.sshd.server.command.CommandFactory;
+import org.apache.sshd.server.keyprovider.SimpleGeneratorHostKeyProvider;
+import org.apache.sshd.server.channel.ChannelSession;
+
+import java.io.IOException;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.security.PublicKey;
+import java.util.List;
+
+public class SSHServerSetup {
+
+    private SshServer sshd;
+
+    public void start() throws Exception {
+        sshd = SshServer.setUpDefaultServer();
+        SSHConfig sshConfig = new SSHConfig();
+
+        int port = sshConfig.port.asInt();
+        String authMethod = sshConfig.auth_method.asString();
+        Path allowedKeysPath = Path.of(sshConfig.allowed_keys_path.asString());
+        Path serverPrivateKeyPath = Path.of(sshConfig.server_private_key.asString());
+        String username = sshConfig.username.asString();
+        String password = sshConfig.password.asString();
+
+        try {
+            setupServer(port, authMethod, allowedKeysPath, serverPrivateKeyPath, username, password);
+            sshd.start();
+            AL.info("SSH server started on port " + port + " with auth method: " + authMethod);
+        } catch (Exception e) {
+            AL.warn("Failed to start SSH server!", e);
+        }
+    }
+
+    public void close() throws IOException {
+        if (sshd != null) {
+            sshd.close();
+            AL.info("SSH server closed.");
+        }
+    }
+
+    public void stop() throws IOException {
+        if (sshd != null) {
+            sshd.stop();
+            AL.info("SSH server stopped.");
+        }
+    }
```

Comment:
What's the difference between stop and close?

Reply:
close() i believe ensures that all resources are fully released. i dont believe i call it anywhere

Reply:
removed, it is not used, and should not be used in place of stop according to GPT

Reply:
shouldn't be close then be called inside stop instead, if that's the case?

Reply:
I thought the same thing. This is from GPT:

Calling `close()` within `stop()` restricts flexibility because `close()` permanently releases all server resources, necessitating a complete reinitialization for restarts. By separating `stop()` and `close()`, you can efficiently pause and resume server operations without the overhead of fully rebuilding the server each time.

What would you rather I use?

Reply:
decided to use close() for end-of-run exit hook. stop() for everything else, as it may be started at any given time

Reply:
You can use close inside stop, no worries

Reply:
in that case, i will replace stop with close, cause close is effectively a more in depth stop